src/test/java/picocli/CommandModelTransformersTest.java

```diff
@@ -0,0 +1,203 @@
+package picocli;
+
+import org.junit.*;
+import org.junit.contrib.java.lang.system.ProvideSystemProperty;
+import org.junit.contrib.java.lang.system.RestoreSystemProperties;
+import org.junit.contrib.java.lang.system.SystemErrRule;
+import org.junit.contrib.java.lang.system.SystemOutRule;
+import org.junit.rules.TestRule;
+import org.junit.Test;
+import java.util.Arrays;
+import java.util.concurrent.Callable;
+
+import static org.junit.Assert.*;
+
+public class CommandModelTransformersTest {
+    // allows tests to set any kind of properties they like, without having to individually roll them back
+    @Rule
+    public final TestRule restoreSystemProperties = new RestoreSystemProperties();
+
+    @Rule
+    public final ProvideSystemProperty ansiOFF = new ProvideSystemProperty("picocli.ansi", "false");
+
+    @Rule
+    public final SystemErrRule systemErrRule = new SystemErrRule().enableLog().muteForSuccessfulTests();
+
+    @Rule
+    public final SystemOutRule systemOutRule = new SystemOutRule().enableLog().muteForSuccessfulTests();
+
+    @CommandLine.Command(name = "aa")
+    private static class AA implements Callable<Integer> {
+        public Integer call() {
+            return 0;
+        }
+    }
+
+
+    @CommandLine.Command(name = "c")
+    private static class C implements Callable<Integer> {
+        public Integer call() {
+            return 0;
+        }
+    }
+
+    @CommandLine.Command(name = "b")
+    private static class B implements Callable<Integer> {
+        public Integer call() {
+            return 0;
+        }
+    }
+
+
+    @CommandLine.Command(name = "a", subcommands = {AA.class}, modelTransformer = A.AFilter.class)
+    private static class A implements Callable<Integer> {
+
+        private static class AFilter implements CommandLine.IModelTransformer {
+            public CommandLine.Model.CommandSpec transform(CommandLine.Model.CommandSpec commandSpec) {
```

Comment:
If we import `CommandLine.Model.CommandSpec`, it will make the code nicer to read. Maybe also import `CommandLine.Command` and `CommandLine.IModelTransformer`.

Reply:
Ok.